randovania/exporter/hints/credits_spoiler.py

```diff
@@ -69,40 +69,68 @@ def generic_credits(
     standard_pickup_configuration: StandardPickupConfiguration,
     all_patches: dict[int, GamePatches],
     players_config: PlayersConfiguration,
-    namer: HintNamer,
-    pickup_name_format: str = "{}",
-    use_player_color: bool = True,
-) -> dict[str, str]:
+) -> list[tuple[PickupEntry, list[OwnedPickupLocation | str]]]:
+    """
+    Returns the credits in the form of a list of tuples that can be used to create your own formatting.
+    The first tuple element will be a PickupEntry, and the second tuple element a list of location entries.
+    An entry is either an OwnedPickupLocation if the pickup is located somewhere, or a string which
+    is used for special cases (such as starting with the pickup).
+    """
     major_pickup_name_order = {
         pickup.name: index for index, pickup in enumerate(standard_pickup_configuration.pickups_state.keys())
     }
 
-    def sort_pickup(p: PickupEntry) -> tuple[float, str]:
+    def sort_pickup(tup: tuple[PickupEntry, list[OwnedPickupLocation | str]]) -> tuple[float, str]:
+        p = tup[0]
         return major_pickup_name_order.get(p.name, math.inf), p.name
 
-    details = get_locations_for_major_pickups_and_keys(all_patches, players_config)
-    major_pickups_spoiler = {
-        pickup: [entry.export(namer, use_player_color) for entry in entries] for pickup, entries in details.items()
-    }
+    details: dict[PickupEntry, list] = get_locations_for_major_pickups_and_keys(all_patches, players_config)
 
     starting_pickups = starting_pickups_with_count(all_patches[players_config.player_index])
     for pickup, count in starting_pickups.items():
         if count < 1:
             continue
 
-        if pickup not in major_pickups_spoiler:
-            major_pickups_spoiler[pickup] = []
+        if pickup not in details:
+            details[pickup] = []
 
         msg = "As a random starting item"
         if count > 1:
             msg += f" ({count} copies)"
-        major_pickups_spoiler[pickup].append(msg)
+        details[pickup].append(msg)
+
+    for pickup, entries in details.items():
+        if not entries:
+            entries.append("Nowhere")
```

Comment:
can you update some test case to hit this?

Reply:
I mostly added it cause the old code had the same thing.
But honestly, I don't even know *how* it can be hit. 
The `details` dict provided by `get_locations_for_major_pickups_and_keys` only has a pickup key, if it also has an entries value. 
The starting pickups loop above only ever adds a new pickup to the dictionary with one message

So from what I can see, this was some weird failsafe that maybe got triggered in the past that nowadays can never get hit?